stock_location_analytic_purchase/models/purchase_order_line.py

```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Eficent Business and IT Consulting Services S.L.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    @api.multi
+    @api.depends('order_line.account_analytic_id')
+    def _compute_check_analytic_id(self):
+        for po in self:
+            if all(line.account_analytic_id for line in po.order_line):
+                a = all(line.account_analytic_id for line in self.order_line)
+                self.check_analytic_id = True
+            else:
+                self.check_analytic_id = False
```

Comment:
Got this error when installing:
![image](https://user-images.githubusercontent.com/19620251/31379284-17c27ebc-adae-11e7-9baf-af0590b4174a.png)
I think you have to use po.check_analytic_id